tests/test_modules.py

```diff
@@ -75,3 +75,16 @@ def test_duplicate_registration():
     """Registering two things with the same name"""
 
     assert m.duplicate_registration() == []
+
+
+def test_builtin_key_type():
+    """Test that all the keys in the builtin modules have type str.
+
+    Previous versions of pybind11 would add a unicode key in python 2.
+    """
+    if hasattr(__builtins__, "keys"):
+        keys = __builtins__.keys()
+    else:  # this is to make pypy happy since builtins is different there.
+        keys = __builtins__.__dict__.keys()
```

Comment:
Why not simply always go through `__builtins__.__dict__.keys()`? It works in CPython and PyPy.

Reply:
answered in gitter, but I'll add here as well. Python 2 has the following amazing feature
```
    In module __main__:
        __builtins__ is a reference to module __builtin__.
        __builtin__ only exists if you import it.
    In any other module:
        __builtins__ is a reference to module __builtin__'s __dict__.
        __builtin__ only exists if you import it.
```
so we need to handle both cases.

Reply:
_sigh_